api/src/paths/project/{projectId}/publish.test.ts

```diff
@@ -0,0 +1,208 @@
+import chai, { expect } from 'chai';
+import { describe } from 'mocha';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+import * as publish from './publish';
+import * as db from '../../../database/db';
+import * as project_update_queries from '../../../queries/project/project-update-queries';
+import { QueryResult } from 'pg';
+import SQL from 'sql-template-strings';
+
+chai.use(sinonChai);
+
+const dbConnectionObj = {
+  systemUserId: () => {
+    return null;
+  },
+  open: async () => {
+    // do nothing
+  },
+  release: () => {
+    // do nothing
+  },
+  commit: async () => {
+    // do nothing
+  },
+  rollback: async () => {
+    // do nothing
+  },
+  query: async () => {
+    // do nothing
+  }
+};
+
+const sampleReq = {
+  keycloak_token: {},
+  params: {
+    projectId: 1
+  },
+  body: {
+    publish: true
+  }
+} as any;
+
+let actualResult = {
+  id: null
+};
+
+//let actualStatus: number = (null as unknown) as number;
+
+const sampleRes = {
+  status: () => {
+    return {
+      json: (result: any) => {
+        actualResult = result;
+      }
+    };
+  }
+  // ,
+  // send: (status: number) => {
+  //   actualStatus = status;
+  // }
+};
+
+describe('project/{projectId}/publish', () => {
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it('should throw a 400 error when missing request param projectId', async () => {
+    sinon.stub(db, 'getDBConnection').returns({
+      ...dbConnectionObj,
+      systemUserId: () => {
+        return 20;
+      }
+    });
+
+    try {
+      const result = publish.publishProject();
+
+      await result(
+        { ...sampleReq, body: { ...sampleReq.body }, params: { projectId: undefined } },
+        (null as unknown) as any,
+        (null as unknown) as any
+      );
+      expect.fail();
+    } catch (actualError) {
+      expect(actualError.status).to.equal(400);
+    }
+  });
+
+  it('should throw a 400 error when missing request body', async () => {
+    sinon.stub(db, 'getDBConnection').returns({
+      ...dbConnectionObj,
+      systemUserId: () => {
+        return 20;
+      }
+    });
+
+    try {
+      const result = publish.publishProject();
+
+      await result(
+        { ...sampleReq, body: (null as unknown) as any },
+        (null as unknown) as any,
+        (null as unknown) as any
+      );
+      expect.fail();
+    } catch (actualError) {
+      expect(actualError.status).to.equal(400);
```

Comment:
For all of these error tests, we should add an expect for `actualError.message`.  Otherwise its possible that the 400 being caught isn't the one we were testing for (as the code has multiple checks that all throw a 400).